### ISSUE TITLE
TTS Tacotron 2 trained on JSUT generates gibberish sound at the end

### ISSUE THREAD

**Issue by tranctan** [espnet/espnet#2138]
Hi, after generating certain samples using JSUT-Tacotron 2 , i found this unusual behavior. The model sometimes generates gibberish sound at the end.

May i ask if this bug is expected for the current Tacotron 2? Or do we have any way to fix this ?

Thank you a lot!

**Comment by sw005320** [comment]
Are you using a pre-trained model or did you train it by yourself?
Yes, this is a known issue in Tacotraon 2 TTS, but a good model (e.g., espnet pre-trained models) with normal prompts do not often have such issues.

**Comment by tranctan** [comment]
Hi, thanks for coming by !

I used pre-trained model that are trained on JSUT corpus, both Tacotron 2 and ParallelWaveGAN.

**Comment by sw005320** [comment]
OK. What kind of the text input causes such an issue?

**Comment by tranctan** [comment]
Sorry for the late response. Here is what i tried:

`多くの異なる質問をすることで彼と話し始めることができます。 何か面白いものから始めましょう。 言ってみて`

I tried generates this sentence for multiple times (20 times), and half of the times it ends up with gibberish sound at the end.

**Comment by sw005320** [comment]
I don't know whether you can read Japanese, but anyway, the above examples include three sentences.
`。` means the period in Japanse.
I also checked the original JSUT training sentences and they are base on **one** sentence.
So, the pre-trained model may behave irregularly if you throw multiple sentences at the same time.
My suggestion is to generate the sentence one-by-one, instead.

**Comment by tranctan** [comment]
Thanks for your suggestion ! I understand what you mean, but here's another trial, this sentence pattern is as same as the sentences in JSUT corpus:
`計画をたてることとそれを実行する事は、言ってみて。`

After the 3.5 seconds, the gibberish sound is generated.
![1](https://user-images.githubusercontent.com/42478131/87114295-6d233f00-c29a-11ea-9847-3382cdde252f.png)

**Comment by sw005320** [comment]
I see. @kan-bayashi, could you give us some comments?

**Comment by kan-bayashi** [comment]
I could not understand from the figure but it seems that stop prediction is failed.
You can try `use-att-constraint: true` or adjust the `threshold`.

**Comment by tranctan** [comment]
Hi ! Thanks for your suggestion!

I tried playing around with `threshold` and `use-att-constraint`.

The change `use-att-constraint` to `true` seemed not affecting anything.

I tried adjusting the `threshold` and the best value could be **0.045**, it really reduces the chance of gibberish sound generated at the end (although not 100%). However, there's another problem of too-low `threshold`, that is it also sometimes omits the last part of the speech, which is after the `<pau>` token.
